src/BridgedRoom.ts

```diff
@@ -486,7 +486,7 @@ export class BridgedRoom {
         const body: ISlackChatMessagePayload = {
             ...matrixToSlackResult,
             as_user: false,
-            username: user.getDisplaynameForRoom(message.room_id) || matrixToSlackResult.username,
+            username: (await user.getDisplaynameForRoom(message.room_id)) || matrixToSlackResult.username,
```

Comment:
Can this throw horribly, and will we lose the messages?

Reply:
Good call. It cannot since 6e1fa00